assets/js/app.js

```diff
@@ -55,8 +55,43 @@ onLoad(() => {
         grid.addWidget(createWidgetElement(item));
     }
 
+    let loading = false;
+    let total = 0;
+
     for (let widget of grid.getGridItems()) {
-        enableScripts(widget);
+        if (widget.dataset.ajax) {
+            loading = true;
+
+            toggleSpin();
+
+            let id = widget.id.replace("widget_", "");
```

Comment:
id field is available through "gs-id" data attribute. It will always contain the correct widget id.